mirascope/core/base/call_params.py

```diff
@@ -1,6 +1,13 @@
 """This module contains the type definition for the base call parameters."""
 
-from typing_extensions import TypedDict
+from typing import Generic, TypeVar
 
+from typing_extensions import NotRequired, TypedDict
 
-class BaseCallParams(TypedDict, total=False): ...  # pragma: no cover
+from mirascope.core import BaseTool
+
+_BaseToolT = TypeVar("_BaseToolT", bound=BaseTool)
+
+
+class BaseCallParams(Generic[_BaseToolT], TypedDict, total=False):
+    tools: NotRequired[list[_BaseToolT]]
```

Comment:
aha ok I think I actually understand this change better now.

We're using `Mapping` for the input call params, which I believe is correct, and we're now using `BaseCallParams` here to type `call_kwargs`, which will always have at least `tools`.

If this is correct, then we should change this to be `BaseCallKwargs` so it's more clear?

Reply:
Fixed